src/pupil_labs/invisible_lsl_relay/xdf_cloud_time_sync.py

```diff
@@ -1,70 +1,78 @@
-        raise FileNotFoundError(f'The path {path_to_xdf} does not point to a file.')
-    if not path_to_xdf.endswith('.xdf'):
-        raise TypeError(f'The file {path_to_xdf} is not an .xdf file.')
-
-    for path in paths_to_export:
-        # each path to an export should be a valid path
-        if not pathlib.Path(path).is_dir():
-            raise NotADirectoryError(
-                'All paths provided after the path '
-                'to the xdf file must be valid directories.'
-            )
-
-
 def load_files(path_to_xdf, paths_to_export):
     # load the xdf file
     try:
         xdf_head, xdf_data = pyxdf.load_xdf(path_to_xdf)
     except Exception:
-        raise OSError(f"Invalid xdf file {path_to_xdf}")
+        raise IOError(f"Invalid xdf file {path_to_xdf}")
+    # extract all serial numbers from gaze streams
+    xdf_serial_nums = extract_serial_num_from_xdf(xdf_head)
+    # check cloud export paths
     for path in paths_to_export:
         for info_path in pathlib.Path(path).rglob("info.json"):
-            if check_files(xdf_head, info_path):
-                perform_time_alignment(path_to_xdf, os.path.dirname(info_path))
+            if json_serial_in_xdf(info_path, xdf_serial_nums):
+                perform_time_alignment(path_to_xdf, info_path.parent)
 
 
 def perform_time_alignment(xdf_path, recording_dir):
     print(f'perform time alignment between {xdf_path} and {recording_dir}')
     save_files()
 
 
-def check_files(head, path):
-    for x in head:
-        if x['info']['type'][0] == 'Gaze':
-            xdf_camera_serial = x['info']['desc'][0]['acquisition'][0][
-                'world_camera_serial'
-            ][0]
-            with open(path) as file:
-                data = json.load(file)
-                try:
-                    cloud_camera_serial = data['scene_camera_serial_number']
-                    if cloud_camera_serial == xdf_camera_serial:
-                        return True
-                except KeyError:
-                    # this is not a pupil info file
-                    pass
+def json_serial_in_xdf(path_infojson, xdf_serial_nums):
+    with open(path_infojson) as infojson:
+        data_infojson = json.load(infojson)
+        if extract_serial_num_from_infojson(data_infojson) in xdf_serial_nums:
+            return True
     return False
 
 
+def extract_serial_num_from_xdf(xdf_head):
+    camera_serial_nums = []
+    for x in xdf_head:
+        try:
+            if x['info']['type'][0] == 'Gaze':
+                xdf_camera_serial = x['info']['desc'][0]['acquisition'][0][
+                    'world_camera_serial'
+                ][0]
+                camera_serial_nums.append(xdf_camera_serial)
+        except KeyError as e:
+            logger.error("The xdf file does not contain the expected fields."
+                         "Make sure it's been streamed with the pupil invisible"
+                         "lsl relay version 2.1.0 or higher.")
+            raise e
```

Comment:
Instead of logging the error and reraising the existing error, raise a new exception
```suggestion
        except KeyError as e:
            raise KeyError(
                "The xdf file does not contain the expected fields. "
                "Make sure it's been streamed with the Pupil Invisible "
                "LSL Relay version 2.1.0 or higher."
            ) from e
```
This mostly matters for the order in which the messages are shown. If you log the message first and then raise the original error, the user will see the specific key error first and will have to scroll up to read the interpretation of what it means.

By raising a new error, the user is presented with two tracebacks with the high-level error description at the bottom.